src/zenml/pipelines/base_pipeline.py

```diff
@@ -181,8 +191,18 @@ def _verify_arguments(self, *steps: BaseStep, **kw_steps: BaseStep) -> None:
                     f"a pipeline."
                 )
 
+            step_ref_func = type(step)
+
+            if step_ref_func in step_refs:
+                step_refs[step_ref_func]
```

Comment:
In this case, wouldnt it make sense to have the error like you had before: `f"Same argument type ({type(step)}) used for arguments {key} and {prev_key}. Step objects should be unique for each argument."`

Reply:
Yup, but the issue is in the case when the user calls the pipeline function like this trainer(evaluator(), evaluator_2 = evaluator()) we won't be able to return the key.

Reply:
Yes for this case (args) we don't need the full message but for the other case (kwargs) we should have it  I think

Reply:
Okay, I will try to see how to do it. It would mean that I have to initialize a new dict for the kwargs and check if the function is repeated there first and then check for the reference in the args.